cfgmgr/vrfmgr.cpp

```diff
@@ -273,3 +291,170 @@ void VrfMgr::doTask(Consumer &consumer)
         it = consumer.m_toSync.erase(it);
     }
 }
+
+bool VrfMgr::doVrfEvpnNvoAddTask(const KeyOpFieldsValuesTuple & t)
+{
+    SWSS_LOG_ENTER();
+    string tunnel_name = "";
+    const vector<FieldValueTuple>& data = kfvFieldsValues(t);
+    for (auto idx : data)
+    {
+        const auto &field = fvField(idx);
+        const auto &value = fvValue(idx);
+
+        if (field == "source_vtep")
+        {
+            tunnel_name = value;
+        }
+    }
+
+    if (m_evpnVxlanTunnel.empty())
+    {
+        m_evpnVxlanTunnel = tunnel_name;
+        VrfVxlanTableSync(true);
+    }
+
+    SWSS_LOG_NOTICE("Added evpn nvo tunnel %s", m_evpnVxlanTunnel.c_str());
+    return true;
+}
+
+bool VrfMgr::doVrfEvpnNvoDelTask(const KeyOpFieldsValuesTuple & t)
+{
+    SWSS_LOG_ENTER();
+
+    if (!m_evpnVxlanTunnel.empty())
+    {
+        VrfVxlanTableSync(false);
+        m_evpnVxlanTunnel = "";
+    }
+
+    SWSS_LOG_NOTICE("Removed evpn nvo tunnel %s", m_evpnVxlanTunnel.c_str());
+    return true;
+}
+
+bool VrfMgr::doVrfVxlanTableCreateTask(const KeyOpFieldsValuesTuple & t)
+{
+    SWSS_LOG_ENTER();
+
+    auto vrf_name = kfvKey(t);
+    uint32_t vni = 0, old_vni = 0;
+    string s_vni = "";
+    bool add = true;
+
+    const vector<FieldValueTuple>& data = kfvFieldsValues(t);
+    for (auto idx : data)
+    {
+        const auto &field = fvField(idx);
+        const auto &value = fvValue(idx);
+
+        if (field == "vni")
+        {
+            s_vni = value;
+            vni = static_cast<uint32_t>(stoul(value));
+        }
+    }
+
+    old_vni = getVRFmappedVNI(vrf_name);
+    SWSS_LOG_NOTICE("VRF VNI map update vrf %s, vni %d, old_vni %d", vrf_name.c_str(), vni, old_vni);
+    if (vni != old_vni)
+    {
+        if (vni == 0)
+        {
+            m_vrfVniMapTable.erase(vrf_name);
+            s_vni = to_string(old_vni);
+            add = false;
+        }
+        else
+        {
+            m_vrfVniMapTable[vrf_name] = vni;
+        }
+
+    }
+
+    if ((vni == 0) && add)
```

Comment:
What will be this case, where VNI is 0 and is still an 'add' condition?

Reply:
This is to take care of the scenario when VRF is configured but VRF to VNI Mapping is not configured.